native_client/deepspeech.cc

```diff
@@ -666,10 +811,31 @@ DS_SpeechToText(ModelState* aCtx,
   if (status != DS_ERR_OK) {
     return nullptr;
   }
+      
   DS_FeedAudioContent(ctx, aBuffer, aBufferSize);
   return DS_FinishStream(ctx);
 }
 
+char*
+DS_SpeechToTextExtended(ModelState* aCtx,
+                const short* aBuffer,
+                unsigned int aBufferSize,
+                unsigned int aSampleRate)
+{
+  StreamingState* ctx;
+  int status = DS_SetupStream(aCtx, 0, aSampleRate, &ctx);
+  if (status != DS_ERR_OK) {
+    return nullptr;
+  }
+  
+  // Store total audio duration in seconds to generate timing info later
+  aCtx->duration_secs = aBufferSize / SAMPLE_RATE;
```

Comment:
Is there any reason why this implementation depends on knowing the length of the entire audio at first? Output frame `t` always starts at `t * 20ms` (20ms is the stride step in the feature computation), so you can use that to convert frame number to time. And at the very least you could compute the duration in `finishStream`, so that this functionality can also work with the streaming API.